### PR TITLE
#167377484 End to end testing using cypress.io

### PR DESCRIPTION
**Pull request by Merkll**
#### What does this PR do?
This pull request configures end to end testing using cypress.io
#### Description of Task to be completed
 - install cypress
 - add cypress test to circleci.yml
#### How should this be manually tested?
- git clone the branch
- cd into the root folder and run ```npm run cy:open```
- cypress should open and run the test
#### Any Background Context You want to provide?
N/A
#### What are the relevant pivotal tracker stories?

[#167377484](https://pivotaltracker.com/story/show/167377484)

#### Screenshots (if appropriate)
Screenshot 2019-08-26 at 5 44 08 PM

#### Questions:
N/A

### CODE REVIEW

File: tests/integration/app.test.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+describe("Google test", () => {
+    it("visit google and search", () => {
```
Comment (iverenshaguy) [inline tests/integration/app.test.js]: Our tabbing should be 2 spaces not 4

File: tests/integration/app.test.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+        cy.visit("https://google.com");
+        cy.get('[name="q"]').type('Search {enter}')
+        cy.get('[name="q"]').should('have.value', 'Search');
+    });
+});
```
Comment (iverenshaguy) [inline tests/integration/app.test.js]: ???

File: cypress.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+    "integrationFolder": "tests/integration"
```
Comment (iverenshaguy) [inline cypress.json]: Our tabbing should be 2 spaces not 4

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+cypress/
+node_modules/
```
Comment (iverenshaguy) [inline .gitignore]: ???

File: tests/integration/app.test.js (hunk 1 above)
Comment (iverenshaguy) [inline tests/integration/app.test.js]:
```suggestion
    it("visits google and search", () => {
```

**Comment by iverenshaguy** [comment]
I'm not sure your test plan is right since after running cy:open, I had tor un the tests myself.

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    - attach_workspace:
+        at: ~/
+    - run:
+        name: Run Cypress tests
+        command: npm run cy:ci
+
+workflows:
+  build:
+    jobs:
+      # split into install + test jobs when Mac caching is correct
```
Comment (iverenshaguy) [inline .circleci/config.yml]: why?

Comment (iverenshaguy) [reply .circleci/config.yml]: I can't see the install job. Does CircleCI do this by default when the Cypress orb is used?

**Review by iverenshaguy — CHANGES_REQUESTED** [review]

**Comment by iverenshaguy** [comment]
I don't think we can merge this till the circle ci task is done. Please review my comments

**Comment by Merkll** [comment]
> I'm not sure your test plan is right since after running cy:open, I had tor un the tests myself.

The headless mode ```cy:run``` automatically runs all test but you will still need to to click run all test test after which any edit will cause the test to rerun.

File: .gitignore (hunk 4 above)
Comment (Merkll) [reply .gitignore]: The cypress folder is generated on test run

File: .circleci/config.yml (hunk 5 above)
Comment (Merkll) [reply .circleci/config.yml]: cypress orb takes care of that

File: .gitignore (hunk 4 above)
Comment (iverenshaguy) [reply .gitignore]: You don't have a new line at the end of this file

**Comment by iverenshaguy** [comment]
> > I'm not sure your test plan is right since after running cy:open, I had tor un the tests myself.
> 
> The headless mode `cy:run` automatically runs all test but you will still need to to click run all test test after which any edit will cause the test to rerun.

The test plan doesn't reflect this. How does someone like me who has never used Cypress know what to do?

**Review by iverenshaguy — APPROVED** [review]

**Pull request merged**